# Introduction 

Ce jupyter Notebook a pour but:

- d'étudier l’effet de 3 éléments de symétrie différents, chacun pour 1 atome différent du matériau qui m'a été attribué
- de mentionner les équations de symétrie 
- d'y inclure les représentations sur base de jupyter-jsmol

Le matériau qui m'a été attribué est le **CsNdTe2** (mp-11742).

### (0) importation des modules nécessaires

In [40]:
# On importe tout les modules qui nous seront potentiellement utiles
# Certains ne sont pas utiles

from pymatgen.ext.matproj import MPRester
from pymatgen.apps.borg.hive import VaspToComputedEntryDrone
from pymatgen.apps.borg.queen import BorgQueen
from pymatgen.entries.compatibility import MaterialsProjectCompatibility
from pymatgen.analysis.phase_diagram import PhaseDiagram, PDPlotter
from jupyter_jsmol.pymatgen import quick_view
from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
from pymatgen.core.lattice import Lattice
from pymatgen import Lattice, Structure
from pymatgen.symmetry.site_symmetries import *
from pymatgen.symmetry.analyzer import *


# On télécharge le fichier CIF du matériau sur le MP sur base de l'API
with MPRester("NwZmcoOXaAFq0K75qQZ") as m:

    # Structure for material id
    structure = m.get_structure_by_material_id("mp-11742")
    
# On travaille sur la cellule conventionnelle
struct = SpacegroupAnalyzer(structure).get_conventional_standard_structure()

# On obtient la liste des symétries contenues dans le CIF
liste_des_symétries = SpacegroupAnalyzer(struct).get_symmetry_operations()



### (1) opération de symétrie n°1

In [48]:
# On a choisit l'opération de symétrie n°13
s1 = liste_des_symétries[12]

# L'atome choisit est un atome de Cs se trouvant en (0.0, 0.0, 0.5)
atome1 = [0.0, 0.0, 0.5]

# application de l'opération de symétrie
image1 = s1.operate(atome1)

print("----------Données brutes----------")
print("coordonnées avant opération:", atome1)
print("coordonnées après opération:", image1)
print("matrice de rotation appliquée:")
print(s1.rotation_matrix)
print("vecteur de translation appliqué:",s1.translation_vector)


----------Données brutes----------
coordonnées avant opération: [0.0, 0.0, 0.5]
coordonnées après opération: [0.66666667 0.33333333 0.83333333]
matrice de rotation appliquée:
[[1. 0. 0.]
 [0. 1. 0.]
 [0. 0. 1.]]
vecteur de translation appliqué [0.66666667 0.33333333 0.33333333]


#### (a) équation de symétrie et commentaires

- L'opération de symétrie **n°13** est une **translation**.
- L'équation de symétrie est **(x,y,z) -> (x + 2/3, y + 1/3, z + 1/3)**.
- Par exemple, l'atome de Cs (n°1) initialement en **(0.0, 0.0, 0.5)**, c'est à dire **(0.0, 0.0, 13.42932)** en coordonnées cartésiennes se retrouve en **(0.6667, 0.333, 0.833)** ; soit en  **(2.34165, 1.35195, 22.38220)** en coordonnées cartésiennes, ce que confirme l'illustration ci-dessous.
- La matrice de rotation est bien unitaire, ce qui prouve que l'opération de symétrie est bien une translation.

#### (b) visualisation de la symétrie

In [57]:
view = quick_view(struct)
display(view) 
view.script('draw symop 13 {atomno=1}')

# remarque: si la symétrie ne s'affiche pas, il faut le faire "manuellement"
# clique droit sur l'atome en question -> symmetry -> choissir le bon numéro. 

JsmolView(layout=Layout(align_self='stretch', height='400px'))

### (2) opération de symétrie n°2

In [52]:
# On a choisit l'opération de symétrie n°2
s2 = liste_des_symétries[1]

# L'atome choisit est un atome de Cs se trouvant (0.33333, 0.66667, 0.16667)
atome2 = [0.33333, 0.66667, 0.16667]

# application de l'opération de symétrie
image2 = s2.operate(atome2)

print("----------Données brutes----------")
print("coordonnées avant opération:", atome2)
print("coordonnées après opération:", image2)
print("matrice de rotation appliquée:")
print(s2.rotation_matrix)
print("vecteur de translation appliqué:",s2.translation_vector)

----------Données brutes----------
coordonnées avant opération: [0.33333, 0.66667, 0.16667]
coordonnées après opération: [-0.33333 -0.66667 -0.16667]
matrice de rotation appliquée:
[[-1.  0.  0.]
 [ 0. -1.  0.]
 [ 0.  0. -1.]]
vecteur de translation appliqué: [0. 0. 0.]


#### (a) équation de symétrie et commentaires

- L'opération de symétrie **n°2** est une **inversion (par rapport à l'origine)**.
- L'équation de symétrie est **(x, y, z) -> (-x, -y, -z)**.
- Par exemple, l'atome de Cs initialement en **(1/3, 2/3, 1/6)**, c'est à dire **(0.00000   2.70390   4.47644)** en coordonnées cartésiennes se retrouve en **(-1/3, -2/3, -1/6)** ; soit en  **(0.00000, -2.70390, -4.47644)** en coordonnées cartésiennes, ce que confirme l'illustration ci-dessous.
- La matrice de rotation est bien l'opposée de la matrice unitaire, et le vecteur de translation est nul, ce qui prouve que l'opération de symétrie est bien une inversion par rapport à l'origine.

#### (b) visualisation de la symétrie

In [55]:
view = quick_view(struct)
display(view) 
view.script('draw symop 2 {atomno=1}')

# remarque: si la symétrie ne s'affiche pas, il faut le faire "manuellement"
# clique droit sur l'atome en question -> symmetry -> choissir le bon numéro.

JsmolView(layout=Layout(align_self='stretch', height='400px'))

### (3) opération de symétrie n°3

In [65]:
# On a choisit l'opération de symétrie n°12
s3 = liste_des_symétries[11]

# L'atome choisit est un atome de Te se trouvant en (1.00000   0.00000   0.73160)
atome3 = [1.00000, 0.00000, 0.73160]

# application de l'opération de symétrie
image3 = s3.operate(atome3)

print("----------Données brutes----------")
print("coordonnées avant opération:", atome3)
print("coordonnées après opération:", image3)
print("matrice de rotation appliquée:")
print(s3.rotation_matrix)
print("vecteur de translation appliqué:",s3.translation_vector)

----------Données brutes----------
coordonnées avant opération: [1.0, 0.0, 0.7316]
coordonnées après opération: [1.     1.     0.7316]
matrice de rotation appliquée:
[[ 1.  0.  0.]
 [ 1. -1.  0.]
 [ 0.  0.  1.]]
vecteur de translation appliqué: [0. 0. 0.]


#### (a) équation de symétrie et commentaires

- L'opération de symétrie **n°12** est un **plan mirroir**.
- L'équation de symétrie est **(x,y,z) -> (x, x-y, z)**.
- Par exemple, l'atome de Te initialement en **(1.0   0.0  0.73160)**, c'est à dire **( 4.68330   0.00000  19.64968)** en coordonnées cartésiennes se retrouve en **(1.     1.     0.7316)** ; soit en  **(4.68330   4.68330  19.64968)** en coordonnées cartésiennes, ce que confirme l'illustration ci-dessous.
- On apercoit bien le plan mirroir en vert grâce à l'illustration ci-dessous.

#### (b) visualisation de la symétrie

In [63]:
view = quick_view(struct)
display(view) 
view.script('draw symop 12 {atomno=1}')

# remarque: si la symétrie ne s'affiche pas, il faut le faire "manuellement"
# clique droit sur l'atome en question -> symmetry -> choissir le bon numéro.

JsmolView(layout=Layout(align_self='stretch', height='400px'))